pkg/virt-controller/watch/vmi.go

```diff
@@ -708,7 +708,8 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 		}
 
 	case vmi.IsRunning():
-		if !vmiPodExists {
+		if !vmiPodExists || isPodDownOrGoingDown(pod) {
```

Comment:
This is wrong. In a normal situation at this point, the owner of the VMI is the handler and only the handler should move the VMI to failed. Here we need to express, "Pod is down, the handler is down, and let's move the VMI to Failed".

If we do not do this then 99% of the time the controller and handler would be racing.

Reply:
I think changing this to `podIsDown` might do the trick. What is left to do is to see if migration doesn't influence this.

Reply:
Done. Please, check if I got it right. Thanks!